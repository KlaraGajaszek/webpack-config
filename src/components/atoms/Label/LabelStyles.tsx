import styled from 'styled-components';

export const LabelTitle = styled.span<{
  color: string;
}>`
  color: ${(props) => props.color};
  font-weight: 600;
`;

export const LabelBox = styled.div`
  display: flex;
  flex-direction: row;
  margin-top: 10px;
`;

export const LabelIcon = styled.label`
  color: #f92a70;
  margin: 0px 5px 0 0;
`;
