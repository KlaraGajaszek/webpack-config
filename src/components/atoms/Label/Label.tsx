import React, { FC } from 'react';

import { LabelTitle, LabelBox, LabelIcon } from './LabelStyles';

type LabelProps = {
  label: string;
  icon?: React.ReactNode;
  color: string;
};

const Label: FC<LabelProps> = ({ icon, label, color }) => {
  return (
    <LabelBox data-testid="labelID">
      <LabelIcon data-testid="labelIconID"> {icon}</LabelIcon>
      <LabelTitle data-testid="labelTitleID" color={color}>
        {label}
      </LabelTitle>
    </LabelBox>
  );
};

export default Label;
