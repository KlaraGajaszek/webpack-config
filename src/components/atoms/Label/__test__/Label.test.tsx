import React from 'react'
import 'jest-styled-components'

import { screen, render, within, getByTestId } from '@testing-library/react'
import renderer from 'react-test-renderer'
import { FaUserAlt } from 'react-icons/fa'

import Label from '../Label'

describe('Label', () => {
    it('render without crash', () => {
        render(<Label label="test" color="red" />)
        const label = screen.getByTestId('labelID')
        expect(label).toBeInTheDocument()
    })
    it('renders correctly', () => {
        const label = renderer
            .create(<Label label="title" icon={FaUserAlt} color="red" />)
            .toJSON()
        expect(label).toMatchSnapshot()
    })
    it('allows to pass styles as props', () => {
        render(<Label color="red" label="test" />)
        const labelTitle = screen.getByTestId('labelTitleID')
        expect(labelTitle).toHaveStyleRule('color', 'red')
    })
})
