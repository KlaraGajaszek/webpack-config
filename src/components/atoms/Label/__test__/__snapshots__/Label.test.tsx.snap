// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders correctly 1`] = `
.c2 {
  color: red;
  font-weight: 600;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-top: 10px;
}

.c1 {
  color: #f92a70;
  margin: 0px 5px 0 0;
}

<div
  className="c0"
  data-testid="labelID"
>
  <label
    className="c1"
    data-testid="labelIconID"
  >
     
  </label>
  <span
    className="c2"
    color="red"
    data-testid="labelTitleID"
  >
    title
  </span>
</div>
`;
